Libplanet/Store/BlockSet.cs

```diff
@@ -49,21 +51,57 @@ public override ICollection<Block<T>> Values
                     throw new KeyNotFoundException();
                 }
 
-                Trace.Assert(block.Hash == key);
+                Trace.Assert(block.Hash.Equals(key));
                 block.Validate(DateTimeOffset.UtcNow);
                 return block;
             }
 
             set
             {
-                if (value.Hash != key)
+                if (!value.Hash.Equals(key))
                 {
                     throw new InvalidBlockHashException(
                         $"{value}.hash does not match to {key}");
                 }
 
                 value.Validate(DateTimeOffset.UtcNow);
-                Store.PutBlock(value);
+
+                byte[] prevMask;
+                if (value.PreviousHash is HashDigest<SHA256> prevHash)
+                {
+                    if (Store.GetAddressesMask(prevHash) is Address maskAddr)
+                    {
+                        prevMask = maskAddr.ToByteArray();
+                    }
+                    else
+                    {
+                        // If there is no mask for the previous hash
+                        // (the store made in the older versions of Libplanet
+                        // may lack mask files), make the mask to match to
+                        // all possible addresses.
+                        prevMask = Enumerable.Repeat((byte)0xff, Address.Size)
```

Comment:
We can also use [WildcardMask](https://github.com/planetarium/libplanet/pull/197/files#diff-f9c16aaa8d3a312223ca86876f791202R21) here, right?

Reply:
`WildcardMask` is defined in the `BlockChain<T>` class and it's `private`.  Making `BlockSet<T>` depends on `BlockChain<T>` also seems unnatural.  So I'm going to define the same constant here too.

It's like we should define a distinct type like `AddressMask` in the future.

Reply:
Hmm, as `BlockChain<T>` already has depended on `BlockSet<T>` (though the inverse is not true), I'm going to move the `WildcardMask` constant to `BlockSet<T>` and make it `internal` instead of `private`.

Reply:
Addressed and amended the commit: 23170446098fc02655e7127b4fc45e8b52e6b7cd.